Assets/Scripts/PlayerMovement.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+/// <summary>
+/// Controls the movement of a player object. Adds gravity and handles move inputs.
+/// </summary>
+[RequireComponent(typeof(CharacterController))]
+public class PlayerMovement : MonoBehaviour
+{
+    [SerializeField] private float moveSpeed;
+    [SerializeField] private float terminalVelocity;
+    private CharacterController cc;
+    private InputAction moveAction;
+    private Vector3 moveVector;
+
+    void Awake()
+    {
+        cc = GetComponent<CharacterController>();
+    }
+
+    void Start()
+    {
+        // returns Vector2 where x = left/right & y = up/down, on controller/keyboard
+        moveAction = InputSystem.actions.FindAction("Move");
+    }
+
+    void Update()
+    {
+        // handle walking movement (horizontal)
+        Vector2 moveInput = moveAction.ReadValue<Vector2>() * moveSpeed;
+        Vector2.ClampMagnitude(moveInput, moveSpeed);
+
+        // transform.forward and transform.right are forward/back & left/right motion respectively
+        // ~in relation to character rotation
+        moveVector = transform.forward * moveInput.y + transform.right * moveInput.x
+            + new Vector3(0, moveVector.y, 0);  // keep Y value the same as last frame
+
+        if (cc.isGrounded)
+            moveVector.y = 0;
+        else
+            // add gravity acceleration~ multiplying by deltaTime twice is NOT a mistake!!
+            moveVector += Physics.gravity * Time.deltaTime;
+
+        cc.Move(moveVector * Time.deltaTime);
```

Comment:
Nice, thank you for thinking of the less fortunate players! `Time.deltaTime` makes sure that those with higher frame rates don't have potentially better odds (or worse depending on the situation) whilst playing.